src/Symfony/Component/EventDispatcher/ContainerAwareEventDispatcher.php

```diff
@@ -105,7 +105,7 @@ public function removeListener($eventName, $listener)
     public function hasListeners($eventName = null)
     {
         if (null === $eventName) {
-            return (bool) count($this->listenerIds) || (bool) count($this->listeners);
+            return $this->listenerIds || $this->listeners;
```

Comment:
Maybe better is to write clear and optimized code and use comments and annotations for easy understanding?

I think, some constructions like as `return 0 < count($this->listenerIds) || 0 < count($this->listeners)` is not easier to understanding, or it maybe easier, for people, who do not know php enough well.

I believe, to use code constructions that have excess of logic is not correct way to build great framework.